### ISSUE TITLE
About pre-trained checkpoints

### ISSUE THREAD

**Issue by jianjun0407** [WinstonHuTiger/mamba_mae#2]
Thank you for your sharing. I have tried your method, and now I have encountered some problems, please advise me.
First of all, I downloaded your open source three model parameters, and I visualized the MAE part  using the code: visualize.py. The results are as follows ( test image, masked test image, network output image, and the masked part of the test image is backfilled with network prediction results): 
![11111111111](https://github.com/user-attachments/assets/44cb2f4f-5cd3-4278-a841-9a7d2741de3e)

From the test results, checkpoint-999_patchSize_16.pth cannot reconstruct the test image. I think this is the problem I use, so I would like to ask:
(1) What data processing needs to be done for the test image sent into the network?
(2) Referring to what you wrote in the readme, my current data processing process is: first crop the foreground area, then resize it to [160,160,160], and then normalize it by min-max, and then multiply it by 255. Finally, the processed data is used to call visualize.py. May I ask if this is the right way to process the image? Given the current test results, where do you think my problem occurred?

![222222](https://github.com/user-attachments/assets/b9767442-9e49-4660-8fa2-f634f5d16ef5)

**Comment by WinstonHuTiger** [comment]
May I ask what data you used and it seems the masking ratio is not right. I am not sure that the model can reconstruct the image for masking ratios other than 0.75.

**Comment by jianjun0407** [comment]
Thank you for your attention, and I will repeat the above operations in detail below, mainly including data processing, testing and visualization process. Please also take a look at my current processing and any problems:
(1) I used the data of the BraTs2023 challenge, which should be similar to the data of the BraTs2022 challenge you used. In the following, take the "BraTs-GLI-00000-000" data as an example.
(2) Refer to the readme, this section of data processing requirements:
![222222](https://github.com/user-attachments/assets/36b7bd72-1ad2-4bd0-a927-468c1813f643)

My approach is to crop out the foreground area of the image, then resize to [160,160,160], then use the maximum-minimum normalization method, zoom to [0,1], then multiply by 255, and finally save the processed data in xxx.npy format.

(3) When visualizing, The "checkpoint-999_patchSize_16.pth" model is used, with mask_ratio=0.75. Here is the code for the visualization section, along with the results :
![44444444](https://github.com/user-attachments/assets/f27ad4b0-c4e4-4a3b-8e1f-bf02e3aa3d69)
![55555](https://github.com/user-attachments/assets/033166de-7619-4019-a843-dd6a51e40aba)

Please help me to look at the use of this checkpint where my current problem occurs? Or, what about the.NPY process you suggest for converting match data to code use ?

**Comment by WinstonHuTiger** [comment]
Sorry, I might made a mistake on the dataset description there.  
It should be brats2021 dataset, but 2022 version and 2021 version are the same.
I can share with you the file where we store our preprocessing records. 

Here is one example visualization of the reconstruction in EGD dataset using the provided checkpoint. 

![image](https://github.com/user-attachments/assets/a4ec4841-54e3-4327-985d-71e91e358363)

I think you have to store the file in ```nii.gz```'s format.

**Comment by jianjun0407** [comment]
Thank you for your reply, I think my problem may appear in the image coordinate system：
(1) My image, which is opened in itk-snap as shown in the following figure, has different positions in the cross, sagittal and coronal planes from your example image above :
![999999999](https://github.com/user-attachments/assets/9d273063-a4d4-4031-a099-f581a6acc4a1)

(2) Can you upload the code for the data processing part? 
For example, if you take data from the brats2021 challenge and read in the .NIi.gz data, what do you need to do to start training with main_pretrain.py?
Specifically, I see that the dataLoader part of main_pretrain.py reads the code as follows, which converts the data of brats2021 challenge to the data that can be processed by the following code. The code of this part can be uploaded ? thank you~

![88888888](https://github.com/user-attachments/assets/1c433467-f1e9-49d8-804a-c71d660daa4c)

**Comment by WinstonHuTiger** [comment]
> Thank you for your reply, I think my problem may appear in the image coordinate system： (1) My image, which is opened in itk-snap as shown in the following figure, has different positions in the cross, sagittal and coronal planes from your example image above : ![999999999](https://private-user-images.githubusercontent.com/19251556/378294787-9d273063-a4d4-4031-a099-f581a6acc4a1.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA0ODc3NTcsIm5iZiI6MTczMDQ4NzQ1NywicGF0aCI6Ii8xOTI1MTU1Ni8zNzgyOTQ3ODctOWQyNzMwNjMtYTRkNC00MDMxLWEwOTktZjU4MWE2YWNjNGExLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAxVDE4NTczN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEzZmJkZTA4MTc0MWNkNDgwYTYwMGI1ZDYxM2JkNTllZTA3NjdmNTRkNWIzY2NmOWE2MjNkN2M3MjkxNDQ0ZDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PBt-pjNeHrvO1dw3R0CA9EZJHHAfZZbqjw9Wgsmgp2Q)
> 
> (2) Can you upload the code for the data processing part? For example, if you take data from the brats2021 challenge and read in the .NIi.gz data, what do you need to do to start training with main_pretrain.py? Specifically, I see that the dataLoader part of main_pretrain.py reads the code as follows, which converts the data of brats2021 challenge to the data that can be processed by the following code. The code of this part can be uploaded ? thank you~
> 
> ![88888888](https://private-user-images.githubusercontent.com/19251556/378294861-1c433467-f1e9-49d8-804a-c71d660daa4c.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA0ODc3NTcsIm5iZiI6MTczMDQ4NzQ1NywicGF0aCI6Ii8xOTI1MTU1Ni8zNzgyOTQ4NjEtMWM0MzM0NjctZjFlOS00OWQ4LTgwNGEtYzcxZDY2MGRhYTRjLlBORz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTAxVDE4NTczN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFiZTNjNGJjNTg4YWJiYjExYmYyMzk5NzcwMmRkOWYzNzNjM2Q5NDg3NjZjYjQxMTQyM2U3YTVlMjYzMDg4MTAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.E0n15OxEZbzckEmEBuoOzOEsdOk4THaczIinwJeoad0)

Sorry for the late response, I was busy with some other stuff. 

I can provide one version here I used to pre-process the TCGA dataset, since I did not write the pre-processing part for Brats or EGD dataset (Dr. Benedikt Wiestler did this part). However, I think it should be the same for all the other datasets as well. The key thing is to remove the affine matrics in volumes and get the center of mass of the volume. After preprocessing, you can save the normalized array to npy file for your model training.

```python
def process_mri_image(input_path):
        # Step 1: Read a 3D MRI image and get the array
        image = sitk.ReadImage(input_path)
        image_array = sitk.GetArrayFromImage(image)
        
        # Step 2: Zero-pad the image to be 320x320x320
        padding = [(max(0, 160 - image_array.shape[i] // 2), max(0, 160 - image_array.shape[i] // 2)) for i in range(3)]
        padded_array = np.pad(image_array, padding, mode='constant', constant_values=0)
        
        # Step 3: Apply a simple threshold to get a rough brain mask
        threshold_value = np.mean(padded_array[padded_array > 0])
        brain_mask = padded_array > threshold_value
        
        # Step 4: Compute the center of mass of the mask
        com = center_of_mass(brain_mask)
        center_index = [int(c) for c in com]
        
        # Step 5: Crop the image to 160x160x160 centered at the center of mass
        
        start = [max(0, center_index[i] - self.size // 2) for i in range(3)]
        end = [start[i] + self.size for i in range(3)]
        cropped_array = padded_array[start[0]:end[0], start[1]:end[1], start[2]:end[2]]
        
        # Step 6: Normalize to [0, 255] with uint8 format
        normalized_array = exposure.rescale_intensity(cropped_array, out_range=(0, 255)).astype(np.uint8)
        
        # Step 7: Return the normalized array
        return normalized_array

```
